HealthMonitoring.SelfHost/Entities/EndpointRegistration.cs

```diff
@@ -14,5 +14,12 @@ public class EndpointRegistration
         public string Group { get; set; }
         [CustomValidation(typeof(TagsValidator), "CheckForUnallowedSymbols")]
         public string[] Tags { get; set; }
+        [MinLength(64)]
+        public string PrivateToken { get; set; }
+
+        public string GetNaturalKey()
```

Comment:
It would be better to not have this method implemented in many places.

Maybe it would be better to change the `EndpointRegistry.GetByNaturalKey()` to accept the monitorType and address parameters, and change it implementation to use the `EndpointIdentity.GetNaturalKey()` under the hood, by either instantiating the EndpointIdentity instance or reworking it a bit to allow getting the key in static context.